### ISSUE TITLE
Hub Packages | Validate Group Names

### ISSUE THREAD

**Issue by EstebanBorai** [infinyon/fluvio#3258]
We must make sure identifiers are fixed to use the correct casing, and also
notify users about this change which occurs under the hood.

This issue is associated to user experience of current product. InfinyOn Cloud users who are unable to get value from existing connectors will need to build their own and they will encounter errors without this fix.

## Scenario

When creating a package and specifying a Group Name we can provide
texts with any casing:

```
cdk generate
🤷   Project Name: TestingConnector
🤷   Please set a group name: EstebanBorai
✔ 🤷   Which type of Connector would you like [source/sink]? · source
✔ 🤷   Will your Connector be public? · true
[1/8]   Done: .gitignore
[2/8]   Done: Cargo.toml
[3/8]   Done: Connector.toml
[4/8]   Done: README.md
[5/8]   Done: config-example.yaml
[6/8]   Done: src/config.rs 
[7/8]   Done: src/main.rs
[8/8]   Done: src
```

`Project Name` gets the kebab case automatically after running this command,
for instance, `TestingConnector ` is replaced with `testing-connector`.

But for group name this is not the case, if we attempt to publish the package
we will see the following:

```
Creating package EstebanBorai/testing-connector@0.1.0
Error: Package verification: group EstebanBorai should be lowercase
group EstebanBorai should be alphanumeric, '-' or '_'
```

**Comment by drc-infinyon** [comment]
@EstebanBorai I see that this is done, can you please add an estimate to this issue and close it.

**Comment by EstebanBorai** [comment]
Resolved on #3264
